src/machines/local_activity_state_machine.rs

```diff
@@ -78,19 +100,32 @@ pub(super) struct ResolveDat {
 pub(crate) enum LocalActivityExecutionResult {
     Completed(Success),
     Failed(ActFail),
+    /// The boolean must be set true if we should not record the marker. EX: If we cancelled before
+    /// dispatching.
+    Cancelled(ActCancel, bool),
```

Comment:
Use named fields instead of simple bool or break into `Cancelled` and `CancelledWithMarker` or something? Doesn't matter to me, just a thought.

Reply:
:+1: